src/main/core/worker.c

```diff
@@ -386,6 +422,44 @@ static void _worker_free(Worker* worker) {
     g_free(worker);
 }
 
+void worker_setMinEventTimeNextRound(SimulationTime simtime,
+                                     SimulationTime currentRoundEndTime) {
+    Worker* worker = _worker_getPrivate();
+    MAGIC_ASSERT(worker);
+
+    int lpi = worker->logicalProcessorIdx;
+
+    // First check if the round changed so we can reset the min time for the LP.
+    // No need to lock: worker is the only one running on lpi right now.
+    if (worker->workerPool->roundEndTimes[lpi] < currentRoundEndTime) {
+        worker->workerPool->roundEndTimes[lpi] = currentRoundEndTime;
+        worker->workerPool->minEventTimes[lpi] = SIMTIME_MAX;
+    }
+
+    // Now update the min time for the LP as needed
+    if (simtime < worker->workerPool->minEventTimes[lpi]) {
+        // We found a new minimum and need to update the min for this LP.
+        // No need to lock: worker is the only one running on lpi right now.
+        worker->workerPool->minEventTimes[lpi] = simtime;
+
+        // Check if we need to update the global time using a (cheaper) read
+        // lock. In most cases this will allow us to avoid the write lock.
+        g_rw_lock_reader_lock(&worker->workerPool->globalTimeLock);
```

Comment:
> A linear scan of integers should be quite fast, and it'd avoid lock contention and accompanying context switches.

This is actually still my slightly preferred approach of all three. It could probably even be vectorized; automatically by the compiler if we're lucky

Reply:
Out of curiosity I poked at this a bit. I think it's not vectorizable after all, but should still be quite fast :)

Reply:
I decided to revert to the linear scan (i.e., just my first commit). It's simpler, and it's not obvious the more complicated thing is better. In any case, phantom won't have an advantage over classic here, so it should be OK even if performance here is non-optimal.